src/sqlfluff/rules/L019.py

```diff
@@ -107,135 +68,19 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
-                        description="Found leading comma. Expected only trailing.",
-                        fixes=[
-                            LintFix.delete(last_leading_comma_seg),
-                            *[
-                                LintFix.delete(d)
-                                for d in memory["whitespace_deletions"]
-                            ],
-                            LintFix.create_before(
-                                anchor_segment=self._get_following_seg(
-                                    context.raw_stack, last_code_seg
-                                ),
-                                edit_segments=[last_leading_comma_seg],
-                            ),
-                        ],
-                    )
-
-        elif self.comma_style == "leading":
-            # A new line preceded by a comma == a trailing comma
-            if context.segment.is_type("newline"):
-                last_seg = self._last_code_seg(context.raw_stack)
-                # no code precedes the current position: no issue
-                if last_seg is None:
-                    return None
-                if last_seg.is_type("comma") and not context.segment.is_templated:
-                    # Trigger fix routine
-                    memory["insert_leading_comma"] = True
-                    # Record where the fix should be applied
-                    memory["last_trailing_comma_segment"] = last_seg
-                    return LintResult(memory=memory)
-            # Have we found a trailing comma violation?
-            if memory["insert_leading_comma"]:
-                # Only insert the comma here if this isn't a comment/whitespace segment
-                if context.segment.is_code:
-                    last_comma_seg = memory["last_trailing_comma_segment"]
-                    # Create whitespace to insert after the new leading comma
-                    new_whitespace_seg = WhitespaceSegment()
-                    return LintResult(
-                        anchor=last_comma_seg,
-                        description="Found trailing comma. Expected only leading.",
-                        fixes=[
-                            LintFix.delete(last_comma_seg),
-                            LintFix.create_before(
-                                anchor_segment=context.segment,
-                                edit_segments=[last_comma_seg, new_whitespace_seg],
-                            ),
-                        ],
-                    )
-        # Otherwise, no issue
-        return None
+        fixes = (
+            ReflowSequence.from_around_target(
+                context.segment,
+                root_segment=context.parent_stack[0],
+                config=context.config,
+            )
+            .rebreak()
+            .get_fixes()
+        )
+
+        if not fixes:
+            return LintResult()
+
+        # TODO: Still need to incorporate the nicer formatted description
```

Comment:
Are you planning to address this "TODO" in the current PR?

Reply:
Good catch, yes. I'll add that in.